descartes_light/solvers/include/descartes_light/solvers/ompl/descartes_space.h

```diff
@@ -0,0 +1,204 @@
+
+        /** \brief This will interpolate to the best of its ability between the two states given.
+         * If t=0 this will return the from input, if t=1 this will return the to input, if from=to then to will be returned.
+         * Any other value of t will result in an interpolation to the first point found under the max_dist_ cost threshhold.
+         * First it is found which side the to state is on relative to the from state, then a search is performed 1 rung to the
+         * side relative to the from state. This adjacent rung is then sequentially searched starting at the idx provided by the
+         * to state, a check if performed to ensure no sampling is attempted outside of the valid range for this searched rung.
+         * Once a valid edge + state with a cost less than the max_dist_ parameter is found, this state is returned. */
+        void interpolate(const ompl::base::State *from, const ompl::base::State *to, double t, ompl::base::State *state) const override;
+
+        ompl::base::StateSamplerPtr allocDefaultStateSampler() const override;
+
+        ompl::base::State *allocState() const override;
+
+        void freeState(ompl::base::State *state) const override;
+
+        void printState(const ompl::base::State *state, std::ostream &out) const override;
+
+        void printSettings(std::ostream &out) const override;
+
+        void setup() override;
+
+        descartes_light::BGLGraph<FloatType> getGraph() const
+        {
+          return graph_;
+        }
+
+        std::vector<std::vector<VertexDesc<FloatType>>> getLadderRungs() const
+        {
+          return ladder_rungs_;
+        }
+
+        double getMaxDist() const
+        {
+          return max_dist_;
+        }
+
+        double getRungToRungDist() const
+        {
+          return rung_to_rung_dist_;
+        }
+
+        double getDistanceEpsilon() const
+        {
+          return distance_epsilon_;
+        }
+
+    protected:
+        /** \brief Boost graph ladder, used to help with edge and vertex cost evaluation */
+        descartes_light::BGLGraph<FloatType> graph_;
+        /** @brief Ladder graph representation of the graph vertices, used for creating edge connections */
+        std::vector<std::vector<VertexDesc<FloatType>>> ladder_rungs_;
+        /** @brief Edge evaluator used to evaluate the cost of edges */
+        const std::vector<typename descartes_light::EdgeEvaluator<FloatType>::ConstPtr> edge_eval_;
+        /** @brief Maximum allowed cost for an edge to be valid */
+        const double max_dist_;
+        /** @brief The cost assigned to calculate a meaningful distance between non-adjacent rungs.
+         * This value should be greater than any possible cost between adjacent rungs, however it should not
+         * be MAX_VALUE because the number of rungs away needs to be meaningful
+         * Example: The cost between rung 1 and rung 5 will be 4*rung_to_rung_dist_ */
+        const double rung_to_rung_dist_;
```

Comment:
This and the `max_dist_` are the most complicated/confusing components of this whole search algorithm. In order to work with the OMPL algorithms we need some measure of distance between any two arbitrary states, so we need to somehow represent that, for [rung,index], [1,2] is closer to [3,4] than [50,2]. This is where `rung_to_rung_dist_` comes into play. To me, it seems that it should also be true that [1,2] is closer to all [2,x] than all [3,y] (I'm actually not certain that this is critical). Therefore, the `rung_to_rung_dist_` should be greater than any adjacent connection. At this moment 10000 is totally arbitrary, but I didn't know what a meaningful default value would be. Theoretically the user should define a meaningful cost given their specific problem.

This same logic applies to the `max_dist_` parameter which should be set to work for each specific problem. `max_dist_` represents the maximum allowed cost for any connection to be valid and that will vary dramatically given each application. I envisioned that likely a user would start with a high `max_dist_` and then see what solution is given and perform further tuning/refinement on the value until more optimal solutions are given.

Given everything I said above, especially the statement of 

> To me, it seems that it should also be true that [1,2] is closer to all [2,x] than all [3,y] (I'm actually not certain that this is critical).

I think maybe it would make sense to make `rung_to_rung_dist_` something like `1000*max_dist_` rather than a uniquely defined value. I believe this should still demonstrate the same behavior of converging on a solution.